etc/docker/kernel-image-puller/kernel_image_puller.py

```diff
@@ -89,7 +89,20 @@ def pull_image(image_name):
 
     Since NotFound exceptions are tolerated, we trap for only that exception and let
     the caller handle others.
+
+    If container runtime is not docker(e.g. containerd), the method returns.
     """
+    if docker_client is None:
+        logger.warning(
+            """
+            KernelInfoPuller only operates on Docker-based K8s systems and is unable to pull kernel images at this time.  
+            Please pre-seed your nodes with the necessary kernel images prior to their use.
+            Image '{}' cannot be pulled.
```

Comment:
Thanks for the update, its appreciated.

I think we should issue the long warning once per iteration (in `fetch_image_names()` or the loop its called from`) and just issue "Image '{}' cannot be pulled."  per image, otherwise its too noisy.

You're missing an f-string or `.format()` suffix.

Reply:
yes, really make sense

Reply:
now i put it in `fetch_image_names` method, and when there is no docker runtime, nothing will be queued to be pulled